File: 13241/13241.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    13241.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2023/12/03 16:33:29 by jihoolee          #+#    #+#              #
#    Updated: 2023/12/03 16:45:33 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_gcd(num1: int, num2: int) -> int:
    if (num2 == 0):
        return num1
    return get_gcd(num2, num1 % num2)


def get_lcm(num1: int, num2: int) -> int:
    return num1 * num2 // get_gcd(num1, num2)


def main() -> None:
    num1, num2 = map(int, input().split())
    print(get_lcm(num1, num2))


if (__name__ == '__main__'):
    main()

File: 13241/13241.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   13241.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/11/12 10:32:23 by jihoolee          #+#    #+#             */
/*   Updated: 2023/11/15 07:22:59 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <cstdint>

int64_t get_gcd(int64_t num1, int64_t num2) {
  if (num2 == 0)
    return num1;
  return get_gcd(num2, num1 % num2);
}

int64_t get_lcm(int64_t num1, int64_t num2) {
  return num1 * num2 / get_gcd(num1, num2);
}

int main(void) {
  int64_t num1, num2;

  std::cin >> num1 >> num2;
  std::cout << get_lcm(num1, num2);
  return 0;
}
